leaderboard/agents/negamax_agent/negamax_agent.py

```diff
@@ -0,0 +1,107 @@
+from random import choice
```

Comment:
Yes, there is a small difference (very subtle, but also very useful).

- With this code, we have a Negamax class and it is possible to parametrize the depth of the search as such: `negamax_depth1 = Negamax(max_depth=1)` and now `negamax_depth1` is a callable and can act as the act function.
- With the [other implementation](https://github.com/j-bernardi/rl-camb-kaggle-connect-x/blob/764463e556c5aea6f61390d2fec83f363510d029/utils/negamax/negamax.py#L6) it is not possible to parametrize the depth. This is the case because a Kaggle agent should only have two parameters: `act(obs, config)` and so a third parameter such as: `negamax_agent(obs, config, depth=2)` can't be used by the framework.

Reply:
ah yep, makes sense to have a seperate file then.